File: go/03_injection.go
// A03 - Injection. VULN: SQL injection via fmt.Sprintf.
package main

import (
	"database/sql"
	"fmt"
)

// VULN: A03
func FindUser(db *sql.DB, name string) (*sql.Rows, error) {
	q := fmt.Sprintf("SELECT * FROM users WHERE name='%s'", name)
	return db.Query(q)
}

File: java/03_injection.java
// A03 - Injection. VULN: SQL injection in PreparedStatement misuse.
import java.sql.*;

public class Injection {
  public static ResultSet findUser(Connection conn, String user) throws SQLException {
    Statement stmt = conn.createStatement();
    // VULN: A03 - concatenated SQL
    return stmt.executeQuery("SELECT * FROM users WHERE name = '" + user + "'");
  }
}
